src/action/payment.js

```diff
@@ -22,18 +22,42 @@ class PaymentAction {
    * @return {undefined}
    */
   listenForUrl(ipc) {
-    ipc.listen('open-url', async (event, url) => {
-      log.info('open-url', url);
-      if (!isLnUri(url)) {
-        return;
-      }
-      while (!this._store.lndReady) {
-        this._tOpenUri = await nap(100);
-      }
-      this.init();
-      this.setAddress({ address: url.replace(PREFIX_URI, '') });
-      this.checkType();
-    });
+    ipc.listen('open-url', (event, url) => this._openUrl(url));
+  }
+
+  /**
+   * Set the listener for the mobile app to handle incoming URIs
+   * containing lightning invoices.
+   * @param  {Object} Linking The expo api to handle incoming uris
+   * @return {undefined}
+   */
+  async listenForUrlMobile(Linking) {
+    Linking.addEventListener('url', ({ url }) => this._openUrl(url));
+    const url = await Linking.getInitialURL();
+    if (!url) {
+      return;
+    }
+    while (!this._store.navReady) {
+      this._tOpenUri = await nap(100);
```

Comment:
Can we replace this line with simply: `await nap(100)`? (and similarly on line `45` and `56`)

Reply:
This is my main comment. if I'm missing something here then feel free to merge.

Reply:
Fixed this in https://github.com/lightninglabs/lightning-app/pull/816/commits/8c69eba48df5c073244d94e2d83cd20f64a377b2